nucypher/cli/commands/bond.py

```diff
@@ -0,0 +1,197 @@
+from nucypher.cli.utils import connect_to_blockchain, get_registry
+from nucypher.config.constants import NUCYPHER_ENVVAR_STAKING_PROVIDER_ETH_PASSWORD
+from nucypher.control.emitters import StdoutEmitter
+from nucypher.crypto.powers import TransactingPower
+
+
+def is_authorized(emitter, staking_provider: ChecksumAddress, agent: PREApplicationAgent) -> None:
+    _authorized = agent.is_authorized(staking_provider=staking_provider)
+    if not _authorized:
+        emitter.message(STAKING_PROVIDER_UNAUTHORIZED.format(provider=staking_provider), color='red')
+        raise click.Abort()
+
+
+def is_bonded(agent, staking_provider, return_address: bool = False) -> Union[bool, Tuple[bool, ChecksumAddress]]:
+    onchain_operator = agent.get_operator_from_staking_provider(staking_provider=staking_provider)
+    result = onchain_operator != NULL_ADDRESS
+    if not return_address:
+        return result
+    return result, onchain_operator
+
+
+def time_elapsed(emitter, agent: PREApplicationAgent, staking_provider: ChecksumAddress) -> None:
+    blockchain = agent.blockchain
+    now = blockchain.get_blocktime()
+    commencement = agent.get_staking_provider_info(staking_provider=staking_provider).operator_start_timestamp
+    min_seconds = agent.get_min_operator_seconds()
+    termination = (commencement + min_seconds)
+    if now < termination:
+        emitter.error(BONDING_TIME.format(date=maya.MayaDT(termination)))
+        raise click.Abort()
+
+
+@click.command('bond')
+@option_registry_filepath
+@option_provider_uri(required=True)
+@option_signer_uri
+@option_operator_address
+@option_staking_provider
+@option_network(required=True)
+@option_force
+def bond(registry_filepath, provider_uri, signer_uri, operator_address, staking_provider, network, force):
+    """
+    Bond an operator to a staking provider.
+    The staking provider must be authorized to use the PREApplication.
+    """
+
+    #
+    # Setup
+    #
+
+    emitter = StdoutEmitter()
+    connect_to_blockchain(provider_uri=provider_uri, emitter=emitter)
+    if not signer_uri:
+        emitter.message('--signer is required', color='red')
+        raise click.Abort()
+    if not network:
+        network = select_network(emitter=emitter)
+
+    signer = Signer.from_signer_uri(signer_uri)
+    transacting_power = TransactingPower(account=staking_provider, signer=signer)
+    registry = get_registry(network=network, registry_filepath=registry_filepath)
+    agent = ContractAgency.get_agent(PREApplicationAgent, registry=registry)
+
+    #
+    # Checks
+    #
+
+    # Check for authorization
+    is_authorized(emitter=emitter, agent=agent, staking_provider=staking_provider)
+
+    # Check bonding
+    if is_bonded(agent=agent, staking_provider=staking_provider, return_address=False):
+        # operator is already set - check timing
+        time_elapsed(emitter=emitter, agent=agent, staking_provider=staking_provider)
```

Comment:
`enough_time_elapsed`(?) as opposed to `time_elapsed`...

Reply:
✔️